Limelight/Stream/Connection.m

```diff
@@ -310,63 +315,86 @@ -(void) terminate
 -(id) initWithConfig:(StreamConfiguration*)config renderer:(VideoDecoderRenderer*)myRenderer connectionCallbacks:(id<ConnectionCallbacks>)callbacks
 {
     self = [super init];
-    
+
     // Use a lock to ensure that only one thread is initializing
     // or deinitializing a connection at a time.
     if (initLock == nil) {
         initLock = [[NSLock alloc] init];
     }
-    
+
     LiInitializeServerInformation(&_serverInfo);
     _serverInfo.address = [config.host cStringUsingEncoding:NSUTF8StringEncoding];
     _serverInfo.serverInfoAppVersion = [config.appVersion cStringUsingEncoding:NSUTF8StringEncoding];
     if (config.gfeVersion != nil) {
         _serverInfo.serverInfoGfeVersion = [config.gfeVersion cStringUsingEncoding:NSUTF8StringEncoding];
     }
-    
+
     renderer = myRenderer;
     _callbacks = callbacks;
-    
+
     LiInitializeStreamConfiguration(&_streamConfig);
     _streamConfig.width = config.width;
     _streamConfig.height = config.height;
     _streamConfig.fps = config.frameRate;
     _streamConfig.bitrate = config.bitRate;
-    
+
+    //This will activate the remote streaming optimization in moonlight-common if needed
+    _streamConfig.streamingRemotely = config.streamingRemotely;
+
+#if TARGET_OS_IPHONE
     // On iOS 11, we can use HEVC if the server supports encoding it
     // and this device has hardware decode for it (A9 and later)
     if (@available(iOS 11.0, *)) {
         // FIXME: Disabled due to incompatibility with iPhone X causing video
         // to freeze. Additionally, RFI is not supported so packet loss recovery
         // is worse with HEVC than H.264.
+
+        //Streaming with a limited bandwith will result in better quality with HEVC
         //_streamConfig.supportsHevc = VTIsHardwareDecodeSupported(kCMVideoCodecType_HEVC);
     }
+#else
+    if (@available(macOS 10.13, *)) {
+        _streamConfig.supportsHevc = true;
```

Comment:
We should still use VTIsHardwareDecodeSupported() here, since not all Macs with 10.13 will have hardware HEVC decode, even if they have a software fallback.

Reply:
My mac is relatively old and doesn't have hardware support, yet it looks much better than H264 on low bitrate connections. My idea would be to either let the user force HEVC or at least force it when the remote streaming optimizations are selected.